syntax = "proto3";

package v1.user;

service User {
    rpc GetPod(GetPodRequest) returns (GetPodResponse);
    rpc PostPod(PostPodRequest) returns (PostPodResponse);
    rpc GetJob(GetJobRequest) returns (GetJobResponse);
}

message PodMessage {
    string namespace = 1;
    string pod_name = 2;
    string ready = 3;
    string status = 4;
    string restart = 5;
    string age = 6;
    string ip = 7;
    string node_name = 8;
}

message GetPodRequest {
    string name_space = 1;
}

message GetPodResponse {
    PodMessage pod_message = 1;
}

message PostPodMessage {
    string yamldata = 1;
}

message PostPodRequest {
    string yaml_data = 1;
}

message PostPodResponse {
    PostPodMessage post_pod_message = 1;
}

message JobMessage {
    string account = 1;
    string node_name = 2;
    string job_state = 3;
    string job_name = 4;
    string start_time = 5;
    string standard_output = 6;
}

message GetJobRequest {
    string node_name = 1;
}

message GetJobResponse {
    JobMessage get_job_message = 1;
}